### ISSUE TITLE
CppTools extension should handle expected timeouts without throwing uncaught exceptions.

### ISSUE THREAD

**Issue by GabeDeBacker** [microsoft/vscode-cpptools#10634]
### Environment

- OS and Version:  Windows 11 Selfhost (latest version)
- VS Code Version:  1.76.0
- C/C++ Extension Version: 1.14.4
- If using SSH remote, specify OS of remote machine: 

### Bug Summary and Steps to Reproduce

Bug Summary:
When executing our extension tests under VSCode's debugger, it constantly hits an uncaught exception due to timeouts (which are actually expected when CppTools calls into our extension)

The implementation of "callTaskWithTimeout" should be authored in a way that doesn't throw uncaught exceptions if the timeout occurs.
```ts
                });
                const configs = yield this.callTaskWithTimeout(provideConfigurationAsync, configProviderTimeout, tokenSource);
                try {

    callTaskWithTimeout(task, ms, cancelToken) {
        let timer;
        const timeout = () => new Promise((resolve, reject) => {
            timer = global.setTimeout(() => {
                clearTimeout(timer);
                if (cancelToken) {
                    cancelToken.cancel();
                }
                reject(localize(27, null, ms));
            }, ms);
        });
        return Promise.race([task(), timeout()]).then((result) => {
            clearTimeout(timer);
            return result;
        }, (error) => {
            clearTimeout(timer);
            throw error;
        });
    }

```

Steps to reproduce:

Expected behavior:

### Configuration and Logs

```shell
N/A
```

### Other Extensions

All extensions disabled except for WAVE for VSCode (Windows IntelliSense Information Provider)

**Comment by Colengms** [comment]
I believe this should have been addressed by: https://github.com/microsoft/vscode-cpptools/pull/10629

**Comment by sean-mcmanus** [comment]
@Colengms Should this go into a 1.14.5 patch update?

**Comment by sean-mcmanus** [comment]
The fix is available with 1.15.0 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.15.0
